From 1dd7aa5fe900b854c03a319fb59b88524e9f7d2d Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Tue, 20 Feb 2024 11:19:52 +0100
Subject: [PATCH 0608/1695] core: mmu: add MEM_AREA_ROM_SEC in check_mem_map()

Handle MEM_AREA_ROM_SEC in check_mem_map() switch case to prevent OP-TEE
core from panicking when such area is mapped.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c329340b3fbd976fb2f4743d7690201e0b3ab556 core: mmu: add MEM_AREA_ROM_SEC in check_mem_map()]
Change-Id: I447db8035237ad6eb08d78da1eb0843dbec3bcd0
Fixes: fc7e0cc38b99 ("core: MEM_AREA_ROM_SEC maps secure read only cached memory")
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
[etienne: picked from a2c1c8e408d3861874236d07618355a63ca65984]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/mm/core_mmu.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/mm/core_mmu.c b/core/mm/core_mmu.c
index 76ae26028..5cf547716 100644
--- a/core/mm/core_mmu.c
+++ b/core/mm/core_mmu.c
@@ -1444,6 +1444,7 @@ static void check_mem_map(struct tee_mmap_region *map)
 		case MEM_AREA_MANIFEST_DT:
 		case MEM_AREA_RAM_SEC:
 		case MEM_AREA_RAM_NSEC:
+		case MEM_AREA_ROM_SEC:
 		case MEM_AREA_RES_VASPACE:
 		case MEM_AREA_SHM_VASPACE:
 		case MEM_AREA_PAGER_VASPACE:
-- 
2.47.2

